<?xml version="1.0"?>

<launch>

  <arg name="namespace" default="vigir/footstep_planning" />
  <group ns="$(arg namespace)">


	<!-- CONFIGURATION -->
  
	<!-- USE TENSORFLOW MODEL OR CONVEX HULL ALGORITHM -->
	<param name="use_tensorflow_model" value="false" type="bool" />
  
	<!-- MODEL PATH -->
	<param name="frozen_model_path" value="$(find multi_contact_point_estimator)/tensorflow/models/frozen_model_three_points.pb" />

  
  
	<!-- PLUGIN DESCRIPTIONS-->

	<!-- MY PLUGIN TERRAIN_MODEL_UNEVEN -->
	<rosparam file="$(find multi_contact_point_estimator)/config/plugin_descriptions.yaml" command="load"/>
    
	<!-- THE DEFAULTS; INCLUDING ORIGINAL TERRAIN_MODEL -->
	<rosparam file="$(find vigir_footstep_planning_default_plugins)/config/plugin_descriptions.yaml" command="load"/>
  
	<!-- THE TWO OTHER-->
	<rosparam file="$(find vigir_footstep_planning_plugins)/config/plugin_descriptions.yaml" command="load"/>

	<!-- PLUGIN SETS -->
	<rosparam file="$(find multi_contact_point_estimator)/config/default_plugin_sets.yaml" command="load"/>


	<include file="$(find vigir_feet_pose_generator)/launch/feet_pose_generator.launch" />
	<include file="$(find vigir_foot_pose_transformer)/launch/foot_pose_transformer.launch" />
	
	<!-- LAUNCH DERIVED VISUALIZATION NODE -->
	<node name="footstep_planning_vis_uneven" pkg="multi_contact_point_estimator" type="footstep_planning_vis_node_uneven" respawn="true" output="screen" />
	
	<!-- LAUNCH DERIVED FOOTSTEP PLANNER -->
	<node name="footstep_planner_uneven" pkg="multi_contact_point_estimator" type="footstep_planner_node_uneven" respawn="true" output="screen">
		<remap from="body_level_grid_map" to="ground_level_grid_map" />
	</node>
	
	<!-- LAUNCH RVIZ WITH OWN PARAMETERS (just small points instead of boxes, nothing important) -->
	<node name="$(anon rviz)" pkg="rviz" type="rviz" args="-d $(find multi_contact_point_estimator)/config/rviz_footstep_planning.rviz" />
	
	<include file="$(find vigir_terrain_classifier)/launch/terrain_classifier_test.launch" />
	

    <!-- specify where to load robot specific parameters -->
    <rosparam file="$(find vigir_footstep_planner)/config/robot_params.yaml" command="load" />
    <param name="params_path" value="$(find vigir_footstep_planner)/config/planner/" />
    <param name="default_params" value="default" />
    
  </group>
</launch>
